mlflow/tensorflow/__init__.py

```diff
@@ -818,9 +866,45 @@ def log_model_without_starting_new_run():
                         tf_meta_graph_tags=[tag_constants.SERVING],
                         tf_signature_def_key="predict",
                     )
-                    save_model(path=local_path, mlflow_model=mlflow_model, **save_model_kwargs)
-                    client = MlflowClient()
-                    client.log_artifacts(_AUTOLOG_RUN_ID, local_path, artifact_path)
+
+                    input_example = None
+                    signature = None
+                    if log_input_examples:
+
+                        def predict_input_fn():
+                            """
+                            Builds an input function to be used for tf's predict
+                            in order to get predicted values required
+                            for the model signature inference.
+                            """
+                            input_slices = input_example_slice
+                            if isinstance(input_example_slice, dict):
+                                input_slices = {
+                                    k: tensorflow.convert_to_tensor(v)
+                                    for k, v in input_example_slice.items()
+                                }
+                            return tensorflow.data.Dataset.from_tensor_slices(input_slices).batch(1)
+
+                        predicted_values = list(tf_model.predict(predict_input_fn))
+
+                        input_example, signature = resolve_input_example_and_signature(
+                            lambda: input_example_slice,
+                            lambda in_ex: infer_signature(input_example_slice, predicted_values[0]),
+                            log_input_examples,
+                            log_model_signatures,
```

Comment:
We also need to apply the logic from https://github.com/mlflow/mlflow/pull/5510/files#diff-e8adca6c94982c5c5bf934b8f9b2226c374a1c97f563eba8b7d1894b1af3a7bbR1044-R1060 so that `log_model_signatures=True` only if `mlflow.tensorflow.autolog(log_model_signatures=True)` was called; it should still be `False` here if `mlflow.autolog(log_model_signatures=True)` was called for backwards compatibility.

Can we extract the logic from https://github.com/mlflow/mlflow/pull/5510/files#diff-e8adca6c94982c5c5bf934b8f9b2226c374a1c97f563eba8b7d1894b1af3a7bbR1044-R1060 into a common function that we can call?